src/scales/scale.linearbase.js

```diff
@@ -15,11 +15,19 @@ function generateTicks(generationOptions, dataRange) {
 	// for details.
 
 	var spacing;
+
 	if (generationOptions.stepSize && generationOptions.stepSize > 0) {
 		spacing = generationOptions.stepSize;
 	} else {
 		var niceRange = helpers.niceNum(dataRange.max - dataRange.min, false);
 		spacing = helpers.niceNum(niceRange / (generationOptions.maxTicks - 1), true);
+
+		var spacingPrecision = generationOptions.precision;
```

Comment:
@etimberg I think it would be better rename the previous `precision` variable to `factor` and reuse it, instead of introducing the variable `spacePrecision`. This to improve readability and as it serves the same purpose later in the function.

```javascript
function generateTicks(generationOptions, dataRange) {
	var ticks = [];
	// To get a "nice" value for the tick spacing, we will use the appropriately named
	// "nice number" algorithm. See http://stackoverflow.com/questions/8506881/nice-label-algorithm-for-charts-with-minimum-ticks
	// for details.

	var factor = 1;
	var spacing;

	if (generationOptions.stepSize && generationOptions.stepSize > 0) {
		spacing = generationOptions.stepSize;
	} else {
		var niceRange = helpers.niceNum(dataRange.max - dataRange.min, false);
		spacing = helpers.niceNum(niceRange / (generationOptions.maxTicks - 1), true);
		if (generationOptions.precision !== undefined) {
			factor = Math.pow(10, generationOptions.precision);
			spacing = Math.ceil(spacing * factor) / factor;
		}
	}
	var niceMin = Math.floor(dataRange.min / spacing) * spacing;
	var niceMax = Math.ceil(dataRange.max / spacing) * spacing;

	// If min, max and stepSize is set and they make an evenly spaced scale use it.
	if (generationOptions.min && generationOptions.max && generationOptions.stepSize) {
		// If very close to our whole number, use it.
		if (helpers.almostWhole((generationOptions.max - generationOptions.min) / generationOptions.stepSize, spacing / 1000)) {
			niceMin = generationOptions.min;
			niceMax = generationOptions.max;
		}
	}

	var numSpaces = (niceMax - niceMin) / spacing;
	// If very close to our rounded value, use it.
	if (helpers.almostEquals(numSpaces, Math.round(numSpaces), spacing / 1000)) {
		numSpaces = Math.round(numSpaces);
	} else {
		numSpaces = Math.ceil(numSpaces);
	}

	if (spacing < 1) {
		factor = Math.pow(10, spacing.toString().length - 2);
		niceMin = Math.round(niceMin * factor) / factor;
		niceMax = Math.round(niceMax * factor) / factor;
	}
	ticks.push(generationOptions.min !== undefined ? generationOptions.min : niceMin);
	for (var j = 1; j < numSpaces; ++j) {
		ticks.push(Math.round((niceMin + j * spacing) * factor) / factor);
	}
	ticks.push(generationOptions.max !== undefined ? generationOptions.max : niceMax);

	return ticks;
}
```

Reply:
I think `precision` is more understandable to end users. E.g. if you want to round to the nearest .01 it's pretty easy to see that this is two decimal places, but I'm not sure we would want to make the user calculate that they should pass in 100

Reply:
@benmccann I am not sure I understand your comment. My proposal wasn't to change the `precision`-option given by the user, but to use the variable `precision` inside the function, instead of introducing an extra variable `spacingPrecision`.

Reply:
Or more precisely renaming the previous use of `precision` to `factor` and use this for both tick spacing as tick rounding, as they both serve the same purpose.

Reply:
I agree with @jcopperfield, it's more understandable renaming `precision` -> `factor` and `spacingPrecision` -> `precision` since these have different units / meanings.

Reply:
I agree, will make this change